### PR TITLE
Bug: Making sure components data is properly joined onto JIRA daily issue field history and downstream models

### PR DESCRIPTION
**Pull request by fivetran-avinash**
## PR Overview
**This PR will address the following Issue/Feature:** #83

**This PR will result in the following new package version:** `v0.14.0`

This is a breaking change as a `dbt run --full-refresh` is required to reflect the proper components data for customers. 

**Please detail what change(s) this PR introduces and any additional information that should be known during the review of this PR:**

The main issue customers are facing is components data was being joined in incorrectly because there would be identifiers in `field_option` that would have the same id as a `component`.

So when the join  below is executed within `jira__daily_issue_field_history`, it returns the field option values instead of the component names for the component fields.

Screenshot 2023-05-10 at 1 29 01 PM

This PR addresses this issue by ensuring we join on components separately from the field option model. The customer must be utilizing the components source model and specifying `components` as a field in the `issue_field_history_columns` in the `dbt_project.yml` for generating historical data for `jira__daily_issue_field_history`. If neither of those conditions hold, components is then removed from `jira__daily_issue_field_history` to avoid any improper joins. See lines 113-116 and lines 127-130 for the modifications in this join.

Screenshot 2023-05-10 at 1 54 33 PM

This conditional is then implemented throughout the model so that there are no issues for customers who choose not to leverage components as well.

### Basic Validation
Please acknowledge that you have successfully performed the following commands locally:
- [x] dbt compile
- [x] dbt run –full-refresh
- [x] dbt run
- [x] dbt test
- [x] dbt run –vars (if applicable)

Before marking this PR as "ready for review" the following have been applied:
- [x] The appropriate issue has been linked and tagged
- [x] You are assigned to the corresponding issue and this PR
- [x] BuildKite integration tests are passing

### Detailed Validation
Please acknowledge that the following validation checks have been performed prior to marking this PR as "ready for review":
- [x] You have validated these changes and assure this PR will address the respective Issue/Feature.
- [x] You are reasonably confident these changes will not impact any other components of this package or any dependent packages.
- [x] You have provided details below around the validation steps performed to gain confidence in these changes.
  
  
**Bug replication**. 
  
Look at `field_id` = 10019 for the `field_option` source model:
Screenshot 2023-05-09 at 10 43 40 PM

Then see the `id` = 10019 for the `component` source model:
Screenshot 2023-05-09 at 10 43 36 PM

Thus the component values returned downstream when the intial join occurs incorrectly returns the field_option values rather than the component names for the components field.

Screenshot 2023-05-09 at 10 29 16 PM
So we need to isolate components with its own specific logic.    
   

**PR solution and implementation**   
With the new join described above in the PR description, we ensure that if components exist, and are one of the fields mentioned in the issue field history table, any join from the existing table on the components field joins on the components table instead of field option and grabs the correct component names.

We made sure our seed files had the proper components information to be joined upstream

Screenshot 2023-05-10 at 3 23 46 PM
Screenshot 2023-05-10 at 3 23 42 PM
Screenshot 2023-05-10 at 3 23 37 PM

As you can see, the new version of the model returns the component value.

Screenshot 2023-05-10 at 12 01 16 AM

Looking upstream at models that flow from the daily issue field history like `jira__issue_enhanced`, you can see components is also being properly brought in. 

Screenshot 2023-05-10 at 12 25 06 AM
We're on the right track!  
  

**What happens if components are not configured or part of field history?**  

Say we remove components from the `issue_field_history_columns`  in our `dbt_project.yml`. That means we should only return `summary` and `story points` if those fields are present in the `issue_field_history`

We added `summary` rows to our seed files to test this (see the above seed files for the intial PR fix.

The resulting `jira__daily_issue_field_history`. was the following after execution.  

Screenshot 2023-05-10 at 1 10 01 PM

You can see that components is now removed after being included in the initial run, but the summary field is populated with the appropriate information.

Similar behavior occurs when the variable `jira_using_components` is set to false.   
  
**What happens on an incremental load?**

Because of the potential sizing of the `jira__daily_issue_field_history` model, many customers opt to do incremental loads to only load the most recent set of updated rows. We tested this by modifying the `int_jira__issue_calendar_spine` to filter on a specific date range so only certain components are returned up to a certain day, then running a `dbt run --full-refresh`.

Screenshot 2023-05-10 at 11 03 05 AM

This would then return records for this issue up to '2020-11-14'.
Screenshot 2023-05-10 at 11 27 32 AM

We then modified the date range by a day ('2020-11-15'), then executed a `dbt run` to see if the components loaded properly.
Screenshot 2023-05-10 at 12 36 58 PM

Screenshot 2023-05-10 at 12 32 05 PM

Just to be safe, we then modified the date range filter in the calendar spine by several days (up to '2020-11-21') to see if there was any issue with multiple dates being incrementally loaded, then executed a `dbt run`.

Screenshot 2023-05-10 at 12 50 35 PM

All looks good!

Running the same steps when components are not being utilized achieved similar results to above--a successful load of the `summary` field by day, with components excluded.   
  
**Non-incremental sense check**

Removing the date_day filters on `int_jira__issue_calendar_spine` and running `dbt run --full-refresh` and `dbt run` on `jira__daily_issue_field_history` yields the full table of results with the correctly populated component name if components are being leveraged.
Screenshot 2023-05-10 at 3 57 20 PM

Likewise, if components are not being leveraged, it removes `components` as a field while still correctly populating `summary`.
Screenshot 2023-05-10 at 3 59 33 PM

### Standard Updates
Please acknowledge that your PR contains the following standard updates:
- Package versioning has been appropriately indexed in the following locations:
    - [x] indexed within dbt_project.yml
    - [x] indexed within integration_tests/dbt_project.yml
- [x] CHANGELOG has individual entries for each respective change in this PR
    
- [NA] README updates have been applied (if applicable)
    
- [NA] DECISIONLOG updates have been updated (if applicable)
- [NA] Appropriate yml documentation has been added (if applicable)

### dbt Docs
Please acknowledge that after the above were all completed the below were applied to your branch:
- [x] docs were regenerated (unless this PR does not include any code or yml updates)

### If you had to summarize this PR in an emoji, which would it be?
 
🐈‍⬛

### CODE REVIEW

**Review by fivetran-joemarkiewicz — CHANGES_REQUESTED** [review]
@fivetran-avinash great job on this PR! I really appreciate the thoughtfulness you put into your submission as it was incredibly helpful when reviewing and understanding the different scenarios that we need to cover with this addition. 🏅 

When reviewing this PR I wanted to look up what components are and how customers leverage them. I came across [this forum post](https://community.atlassian.com/t5/Jira-questions/Choosing-multiple-components-when-creating-a-task/qaq-p/1261130) where a Jira user adds up to 50 components to a single issue 🙀! I noticed we only tested the logic when one component is selected. As such, I wanted to do a quick test on our own internal data (issue_id = 10091, and schema `jira_3`) and decided to add multiple components.

I re-ran the model for an incremental load and saw the incremental logic work as intended! It captured the update today and inserted the new records. However, it looks like it inserted the IDs and not the actual name of the components. It seems that since these components are string_agg'd in an earlier model, the [join](https://github.com/fivetran/dbt_jira/blob/0efed28dd936ea282ad6c4a7d28d5c347aed2ca8/models/jira__daily_issue_field_history.sql#L127-L130) in your PR does not work as expected when there are multiple components. 

image


I did some digging and found [where the string_agg takes place](https://github.com/fivetran/dbt_jira/blob/0efed28dd936ea282ad6c4a7d28d5c347aed2ca8/models/intermediate/field_history/int_jira__agg_multiselect_history.sql#L49). Do you think there is a way we could update the package to account for multiple components? I wonder if adding the component join logic that you have in the `jira__daily_issue_field_hstory` to a cte in the `int_jira__agg_multiselect_history` could do the trick. Open to your thoughts!

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+# dbt_jira v0.14.0
+## 🚨 Breaking Changes 🚨:
+- Fixed `jira__daily_issue_field_history` model to make sure component values are correctly joined into our issue models ([#99](https://github.com/fivetran/dbt_jira/pull/99)) and configure whether to bring in components if the component source is being leveraged and components is specified as one of the `issue_field_history_columns` to bring in. 
+- Added additional seed rows to ensure the new configuration for components properly runs for all edge cases and compare against normal issue field history fields like `summary`. 
+- Please note, a `dbt run --full-refresh` will be required after upgrading to this version in order to capture the updates.
```
[the remark this answers is not in the document]
Comment (fivetran-avinash) [reply CHANGELOG.md:5]: Noted!

**Comment by fivetran-avinash** [comment]
@fivetran-joemarkiewicz The above issue has been addressed to bring in component ids if there are multiple values, but hasn't been addressed to change them back into component names. Additionally, I'm seeing this issue recur for other field values, like sprints. See the below screenshot. 
Screenshot 2023-05-17 at 1 01 53 PM

So, we will need to create a new feature request to handle this particular inconsistency. 

For the purposes of this bug though, it does bring in the proper component values for each issue and day (even if in varying name or id form), so this should close out this particular task.

**Review by fivetran-joemarkiewicz — APPROVED** [review]
lgtm

**Pull request merged**